dockstore-webservice/src/main/java/io/dockstore/webservice/languages/LanguagePluginHandler.java

```diff
@@ -112,29 +114,23 @@ public VersionTypeValidation validateWorkflowSet(Set<SourceFile> sourcefiles, St
             String absolutePath = file.getAbsolutePath();
 
             MinimalLanguageInterface.GenericFileType fileType;
-            switch (file.getType()) {
-            case DOCKSTORE_SMK:
-            case DOCKSTORE_CWL:
-            case DOCKSTORE_WDL:
-            case NEXTFLOW_CONFIG:
-            case NEXTFLOW:
-            case DOCKSTORE_SERVICE_YML:
-            case DOCKSTORE_SERVICE_OTHER:
-            case DOCKSTORE_YML:
+            switch (file.getType().getCategory()) {
```

Comment:
You don't do a null check here, but you do one on line 184. Is that intentional?

Reply:
I think so; if the file type is null or not one listed, then there is an error, and we should throw an exception.

Reply:
This will throw a NullPointerException if it's null. We should try to avoid throwing those.

Reply:
Null check added